cli/src/lockfiles/javascript.rs

```diff
@@ -91,16 +91,24 @@ impl Parseable for YarnLock {
             let resolution = package
                 .get(&"resolution".into())
                 .and_then(YamlValue::as_str)
+                .filter(|s| !s.is_empty())
                 .ok_or_else(|| "Failed to parse resolution field in yarn lock file".to_owned())?;
 
             // Ignore workspace-local dependencies like project itself ("project@workspace:.").
-            if resolution.contains("@workspace:") {
+            if resolution[1..].contains("@workspace:") {
                 continue;
             }
 
-            let (name, _version) = resolution
-                .rsplit_once("@npm:")
-                .ok_or_else(|| "Failed to parse name in yarn lock file".to_owned())?;
+            let (name, _version) = if let Some(index) = resolution[1..].find("@patch:") {
+                // Extract npm version from patched dependencies.
+                resolution[index + 8..]
```

Comment:
Should this be `"@patch:".len()+1`?

Reply:
I can use `"@patch:".len() + 1`, I don't think splitting is necessary when you already know the size.